app/parse.py

```diff
@@ -15,8 +29,62 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def parse_single_product(product: Tag) -> Product:
+    return Product(
+        title=product.select_one(".title")["title"],
+        description=product.select_one(".description").text,
+        price=float(product.select_one(".price").text.replace("$", "")),
+        rating=len(product.select(".ws-icon-star")),
+        num_of_reviews=int(product.select_one(".review-count").text.split()[0])
+    )
+
+
+def parse_single_page(
+        page_url: str, output_csv_path: str, driver: webdriver
+) -> None:
+    driver.get(page_url)
+    try:
+        time.sleep(0.1)
+        driver.find_element(By.ID, "closeCookieBanner").click()
+    except NoSuchElementException:
+        pass
+
+    while True:
+        try:
+            button = driver.find_element(
+                By.CLASS_NAME, "ecomerce-items-scroll-more"
+            )
+            if button.is_displayed():
+                button.click()
+                time.sleep(0.1)
```

Comment:
https://selenium-python.readthedocs.io/waits.html#explicit-waits
try to use WebDriverWait